hibernate-core/src/main/java/org/hibernate/hql/internal/ast/util/JoinProcessor.java

```diff
@@ -167,14 +168,14 @@ public static JoinType toHibernateJoinType(int astJoinType) {
 	private void collectReferencedTables(ASTIterator iterator, Set<String> result) {
 		while ( iterator.hasNext() ) {
 			AST node = iterator.nextNode();
-			if ( node instanceof FromReferenceNode ) {
-				FromReferenceNode fromReferenceNode = (FromReferenceNode) node;
+			if ( node instanceof TableReferenceNode) {
+				TableReferenceNode fromReferenceNode = (TableReferenceNode) node;
 				String[] tables = fromReferenceNode.getReferencedTables();
 				if ( tables != null ) {
 					Collections.addAll(result, tables);
 				}
 			}
-			else if (node instanceof SqlFragment) {
+			if (node instanceof SqlFragment) {
```

Comment:
Would be better to have empty char delimiter above as elsewhere, btw

Reply:
Kept the formatting here as is (not my line originally), but yes, we could make the formatting here consistent.